MockPyth.sol

```diff
@@ -15,11 +16,35 @@ contract MockPyth is AbstractPyth {
     // You can create this data either by calling createPriceFeedData or
     // by using web3.js or ethers abi utilities.
     function updatePriceFeeds(bytes[] memory updateData) public override {
+        uint numStoredPrices = 0;
+
+        // Chain ID is id of the source chain that the price update comes from. Since it is just a mock contract
+        // We set it to 1.
+        int8 chainId = 1;
+
         for(uint i = 0; i < updateData.length; i++) {
             PythStructs.PriceFeed memory priceFeed = abi.decode(updateData[i], (PythStructs.PriceFeed));
-            priceFeeds[priceFeed.id] = priceFeed;
-            emit PriceUpdate(priceFeed.id, priceFeed.publishTime);
+
+            if (priceFeeds[priceFeed.id].publishTime < priceFeed.publishTime) {
+                emit PriceUpdate(priceFeed.id, true, chainId, sequenceNumber, priceFeed.publishTime,
+                    priceFeeds[priceFeed.id].publishTime, priceFeed.price, priceFeed.conf);
+                priceFeeds[priceFeed.id] = priceFeed;
+                numStoredPrices += 1;
+            } else {
+                // The price update is not stored because an update with more recent publish time is already stored.
+                emit PriceUpdate(priceFeed.id, false, chainId, sequenceNumber,
```

Comment:
The only difference between the `PriceUpdate` events emitted in each branch is the `stored` parameter. You could avoid the need for the branch if you held `stored` (or even `fresh`) in a variable and passed this in to the event at the bottom, outside the if statement.

Reply:
+1. thanks!